Avoid false-positive matches on intermediate objects in `ecs@mappings`

felixbarny opened the pull request:
Fixes #102794

felixbarny committed:
Avoid false-positive matches on intermediate objects in ecs@template

elasticsearchmachine commented:
Pinging @elastic/es-data-management (Team:Data Management)

felixbarny committed:
Update docs/changelog/105440.yaml

elasticsearchmachine commented:
Hi @felixbarny, I've created a changelog YAML for you.

axw reviewed (approved):
LGTM, thanks! I forgot to come back after my PR merged :sweat_smile:

felixbarny committed:
Increment REGISTRY_VERSION

ruflin commented on x-pack/plugin/core/template-resources/src/main/resources/ecs@mappings.json:
```diff
@@ -149,27 +156,7 @@
-            },
-            "path_match": "*.usage",
-            "match_mapping_type": "string"
+            "match_mapping_type": ["double", "long", "string"]
```
Nice simplification

ruflin commented on x-pack/plugin/core/template-resources/src/main/resources/ecs@mappings.json:
```diff
@@ -180,7 +167,8 @@
               "*.location"
-            ]
+            ],
+            "match_mapping_type": ["object", "string"]
```
Is this needed?

felixbarny replied:
Let me maybe remove that, as there are more ways to specify a [`geo_point`](https://www.elastic.co/guide/en/elasticsearch/reference/current/geo-point.html) value, that wouldn't covered by this, for example `"location": [ -71.34, 41.12 ]`.

But as we need to include objects in the `match_mapping_type`, it risks false-positive matches for things like `"foo.location.bar":  "foo"`. But `subobjects: false` does come to the rescue here.

felixbarny committed:
Remove `match_mapping_type` for `geo_point` fields

It also needs to include numeric types because
geo points can be expressed as an array with the format: [lon, lat]
